src/cata_imgui.cpp

```diff
@@ -36,8 +36,39 @@ struct pairs {
     short BG;
 };
 
+ImVec4 impalette[256] = {};
 std::array<RGBTuple, color_loader<RGBTuple>::COLOR_NAMES_COUNT> rgbPalette;
-std::array<pairs, 100> colorpairs;   //storage for pair'ed colored
+std::array<pairs, 100> colorpairs;   //storage for paired colors
+
+static ImVec4 compute_color( uint8_t index )
+{
+    if( index < 16 ) {
+        RGBTuple &rgbCol = rgbPalette[index];
+        return { static_cast<float>( rgbCol.Red ) / 255.0f,
+                 static_cast<float>( rgbCol.Green ) / 255.0f,
+                 static_cast<float>( rgbCol.Blue ) / 255.0f,
+                 1.0f };
+    } else if( index < 232 ) {
+        static uint8_t colors[6] = {0, 95, 135, 175, 215, 255};
```

Comment:
I think constexpr may be better here instead of static.

Reply:
I could make them `const`, but I don’t see how `constexpr` would be of any benefit. They would still need to be included in the executable, and even making the whole `compute_color` function `constexpr` would just result in a larger table included in the binary.

Reply:
Part of this is about intent, part of it is to guarantee immutability of the data, part of it is to allow compiler to do whatever optimizations it can do knowing that the variable is immutable at compile time.

Reply:
But the variables are not used at compile time, so adding `constexpr` improves nothing.